src/anndata/_core/sparse_dataset.py

```diff
@@ -588,10 +588,11 @@ def append(self, sparse_matrix: ss.csr_matrix | ss.csc_matrix | CSArray) -> None
         append_data = sparse_matrix.data
         append_indices = sparse_matrix.indices
         if isinstance(sparse_matrix.data, ZarrArray) and not is_zarr_v2():
-            append_data = append_data[...]
-            append_indices = append_indices[...]
-        data[orig_data_size:] = append_data
-
+            append_data_in_memory = append_data[...]
+            data[orig_data_size:] = append_data_in_memory
+            del append_data_in_memory
```

Comment:
why not like this? if the rhs loads into memory, that happens before the `__setitem__`.

```suggestion
            data[orig_data_size:] = append_data[...]
```

Reply:
Just a brain fart!